packages/resilience/src/resilience.spec.ts

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+import nock from 'nock';
+import { executeWithMiddleware } from './middleware';
+import { retry } from './retry';
+import { timeout } from './timeout';
+
+describe('combined resilience features', () => {
+  const HTTP_STATUS = {
+    OK: 200,
+    UNAUTHORIZED: 401,
+    FORBIDDEN: 403,
+    SERVICE_UNAVAILABLE: 504
+  };
+
+  it('Needs to retry with delay below timeout', async () => {
+    nock('https://example.com', {})
+      .get('/retry')
+      .delay(300)
+      .reply(HTTP_STATUS.SERVICE_UNAVAILABLE)
+      .get('/retry')
+      .delay(300)
+      .reply(HTTP_STATUS.OK);
+
+    const request = () =>
+      axios.request({
+        baseURL: 'https://example.com',
+        method: 'get',
+        url: '/retry'
+      });
+
+    await expect(
+      executeWithMiddleware(
+        [timeout(600), retry(2)],
+        { uri: 'https://example.com', args: [] },
+        request
+      )
+    ).resolves.not.toThrow();
+  });
+  it('Needs to retry with delay above timeout', async () => {
+    nock('https://example.com', {})
```

Comment:
[pp] you could change the test slightly so that both mocks return some okish status code but different data e.g. `killded by timeout` and `response ok`. Then you make the call and assert that the data is `response ok`. Doing that you test that the timeout actually killed the first request because return was 200 and the retry was executed getting the data. In the current test you do not really know if the retry stopped the first request or the bad status.

Reply:
yeah this makes sense, thanks for the hint